File: lib/condo_user_roles.dart
/// Perfis retornados pelo login (`app_users.role` na API).
abstract final class CondoUserRoles {
  static const syndic = 'syndic';
  static const administrator = 'administrator';
  static const resident = 'resident';
  static const partner = 'partner';
  static const collaborator = 'collaborator';

  static String labelPt(String role) {
    switch (role) {
      case syndic:
        return 'Síndico';
      case administrator:
        return 'Administração';
      case resident:
        return 'Morador';
      case partner:
        return 'Parceiros';
      case collaborator:
        return 'Colaboradores';
      default:
        return role;
    }
  }

  static bool isBillingStaff(String role) =>
      role == syndic || role == administrator;

  /// Calendário da equipe, cadastro de espaços e aprovação de reservas (síndico e administração).
  static bool canManageReservationSpaces(String role) => isBillingStaff(role);

  /// Área de documentos: envio e exclusão (síndico e administração).
  static bool canManageDocuments(String role) => isBillingStaff(role);

  static bool isOperationalStaff(String role) =>
      role == syndic || role == administrator || role == collaborator;

  /// Role enviada em `/api/individual-comms/staff-sent` e ao redigir como equipe.
  static String? staffMessagingApiRole(String role) =>
      isOperationalStaff(role) ? role : null;

  static bool canOpenAdministrationHub(String role) => isOperationalStaff(role);

  /// Cadastro do quadro e da escala (síndico e administração).
  static bool canManageCollaboratorsBoard(String role) =>
      role == syndic || role == administrator;

  /// Aba «Escala»: síndico, administração e colaboradores (moradores só veem a lista).
  static bool canViewCollaboratorScheduleTab(String role) =>
      role == syndic || role == administrator || role == collaborator;

  /// Cadastro na guia de serviços (síndico, administração e parceiros).
  static bool canManageServiceGuideCatalog(String role) =>
      role == syndic || role == administrator || role == partner;

  /// Mercado interno — aba «do condomínio»: síndico, administração e parceiros.
  static bool canPostMarketplaceCondominium(String role) =>
      role == syndic || role == administrator || role == partner;

  /// Mercado interno — aba «dos moradores»: moradores e síndico.
  static bool canPostMarketplaceResidents(String role) =>
      role == resident || role == syndic;
}

File: frontend/src/condoUserRoles.ts
/** Perfis em `app_users.role` (alinhado ao app móvel). */
export const CondoUserRoles = {
  admin: 'admin',
  syndic: 'syndic',
  administrator: 'administrator',
  resident: 'resident',
  partner: 'partner',
  collaborator: 'collaborator',
  doorman: 'doorman',
} as const;

export function labelPt(role: string): string {
  switch (role) {
    case CondoUserRoles.admin:
      return 'Administrador da plataforma';
    case CondoUserRoles.syndic:
      return 'Síndico';
    case CondoUserRoles.administrator:
      return 'Administração';
    case CondoUserRoles.resident:
      return 'Morador';
    case CondoUserRoles.partner:
      return 'Parceiros';
    case CondoUserRoles.collaborator:
      return 'Colaboradores';
    case CondoUserRoles.doorman:
      return 'Portaria';
    default:
      return role;
  }
}

/** Administrador da plataforma (cadastro global de condomínios, não confundir com «administrator» do condomínio). */
export function isPlatformAdmin(role: string): boolean {
  return role === CondoUserRoles.admin;
}

export function isBillingStaff(role: string): boolean {
  return role === CondoUserRoles.syndic || role === CondoUserRoles.administrator;
}

export function canManageCondoHomeLayout(role: string): boolean {
  return isBillingStaff(role) || isPlatformAdmin(role);
}

export function isOperationalStaff(role: string): boolean {
  return (
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.collaborator ||
    role === CondoUserRoles.doorman
  );
}

export function canOpenAdministrationHub(role: string): boolean {
  return isOperationalStaff(role);
}

export function picksCondoBeforeContact(role: string): boolean {
  return (
    role === CondoUserRoles.admin ||
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.partner
  );
}

export function canManageReservationSpaces(role: string): boolean {
  return isBillingStaff(role);
}

export function canManageDocuments(role: string): boolean {
  return (
    isBillingStaff(role) ||
    role === CondoUserRoles.collaborator ||
    role === CondoUserRoles.doorman ||
    role === CondoUserRoles.partner
  );
}

/** Mercado interno — aba do condomínio: síndico, administração e parceiros. */
export function canPostMarketplaceCondominium(role: string): boolean {
  return (
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.partner
  );
}

/** Mercado interno — aba dos moradores: moradores e síndico. */
export function canPostMarketplaceResidents(role: string): boolean {
  return role === CondoUserRoles.resident || role === CondoUserRoles.syndic;
}

/** Guia de serviços — cadastro no catálogo: síndico, administração e parceiros. */
export function canManageServiceGuideCatalog(role: string): boolean {
  return (
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.partner
  );
}

/** Role usada em `fromStaffRole` / `viewerStaffRole` na API de comunicados individuais. */
export function staffMessagingApiRole(role: string): string | null {
  if (
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.collaborator ||
    role === CondoUserRoles.doorman
  ) {
    return role;
  }
  return null;
}

/** Quadro de colaboradores — cadastro e escala: síndico e administração. */
export function canManageCollaboratorsBoard(role: string): boolean {
  return isBillingStaff(role);
}

/** Aba «Escala» no quadro: síndico, administração e colaboradores (leitura). */
export function canViewCollaboratorScheduleTab(role: string): boolean {
  return (
    role === CondoUserRoles.syndic ||
    role === CondoUserRoles.administrator ||
    role === CondoUserRoles.collaborator ||
    role === CondoUserRoles.doorman
  );
}
